internal/bminventory/inventory.go

```diff
@@ -1567,16 +1568,109 @@ func (b *bareMetalInventory) noneHaModeClusterUpdateValidations(cluster *common.
 }
 
 func (b *bareMetalInventory) UpdateCluster(ctx context.Context, params installer.UpdateClusterParams) middleware.Responder {
-	c, err := b.UpdateClusterInternal(ctx, params)
+	log := logutil.FromContext(ctx, b.log)
+
+	tx := b.db.Begin()
+	txSuccess := false
+
+	defer func() {
```

Comment:
Seems there is logic duplication.  Especially the extract code, recover code and db code